launchprocess.q

```diff
@@ -0,0 +1,45 @@
+/function which execute system commands from q, and is tailored to deal with the exit codes of lproc.sh:
+/cmd is a string and params are the process parameters which are passed to launch() 
+syscomm:{[params;cmd]
+
+    prevexitcode:first `$system cmd,"; echo $?";
+    $[`0~prevexitcode;
+        .lg.o[`launchprocess;"Successful execution: Starting ",params[`procname]];
+      `1~prevexitcode;
+        .lg.e[`launchprocess;"Error: ",params[`procname]," failed to start"];
+      `2~prevexitcode;
+        .lg.e[`launchprocess;"Error: ",params[`procname]," already running"];
+      .lg.e[`launchprocess;"Unknown error encountered, ",params[`procname]," failed to launch"]]   }
+
+
+
+/function which lets us call launchprocess.sh from inside a TorQ process
+/it takes a dictionary of parameters which will be passed to launchprocess.sh, i.e "-procname rdb1 -proctype rdb" etc.
+
+/ not sure how namespace declaration works     \d .process
+
+launch:{ [params]
+    /exit immediately if process name and type aren't provided
+    if[not 11b~`procname`proctype in key params;0N!
+        .lg.e[`launchprocess;"Process name and type must be provided"];
+        :()];
+    
+    /set default values with .Q.def and string the result for the startline:
+    /for now, let the set of parameters which need to be provided with defaults be:
+    /(procname,proctype),U,localtime
+
+    deflt:`procname`proctype`U`localtime!(`;`;`$getenv[`KDBAPPCONFIG],"/passwords/accesslist.txt";1);
+    params:string each .Q.def[deflt] params; 
+   
+    $[4<count params;
+          [ extras:(-1*(count params)-4)#params; 
+            extraargs:{raze "-",string x[0]," ",string x[1]} each (key extras),'{`$x} each value extras];
+           
+          extraargs:" "];
+
+    sline:"bash ",getenv[`TORQHOME],"/launchprocess.sh -procname ",params[`procname]," -proctype ",params[`proctype]," -U ",params[`U]," -localtime ",params[`localtime],raze " ",'extraargs;
```

Comment:
```suggestion
    sline:"bash ",getenv[`TORQHOME],"/launchprocess.sh -procname ",params[`procname]," -proctype ",params[`proctype]," -U ",params[`U]," -localtime ",params[`localtime]," " sv extraargs;
```

(although if taking above suggestion, you'll also remove procname, proctype etc. here)

Reply:
Yeah everything here has been changed but I have used that suggestion of sv, thanks.